openshift-kube-apiserver/admission/customresourcevalidation/dns/validate_dns.go

```diff
@@ -142,3 +145,107 @@ func validateTolerations(versionedTolerations []corev1.Toleration, fldPath *fiel
 	allErrors = append(allErrors, apivalidation.ValidateTolerations(unversionedTolerations, fldPath)...)
 	return allErrors
 }
+
+// validateUpstreamResolversCreate validates configuration of the Upstream objects when TLS is configured.
+func validateUpstreamResolversCreate(upstreamResolvers operatorv1.UpstreamResolvers, fieldPath *field.Path) field.ErrorList {
+	var errs field.ErrorList
+
+	errs = append(errs, validateDNSTransportConfig(upstreamResolvers.TransportConfig, fieldPath)...)
+
+	if upstreamResolvers.TransportConfig.Transport == operatorv1.TLSTransport {
+		// Transport is TLS so we must check if there are mixed Upstream types. SystemResolveConf is not allowed with TLS.
+		for i, upstream := range upstreamResolvers.Upstreams {
+			errPath := fieldPath.Child("upstreams")
+			if upstream.Type == operatorv1.SystemResolveConfType {
+				errMessage := "SystemResolvConf is not allowed when TLS is configured as the transport"
+				errs = append(errs, field.Invalid(errPath.Index(i).Child("type"), upstream.Type, errMessage))
+			}
+		}
+	}
+
+	return errs
+}
+
+// validateUpstreamResolversUpdate validates configuration of the Upstream objects when TLS is configured.
+func validateUpstreamResolversUpdate(newUpstreamResolvers operatorv1.UpstreamResolvers, oldUpstreamResolvers operatorv1.UpstreamResolvers, fieldPath *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	newUpstreams := newUpstreamResolvers.Upstreams
+	oldUpstreams := oldUpstreamResolvers.Upstreams
+	newTransport := newUpstreamResolvers.TransportConfig.Transport
+
+	if !reflect.DeepEqual(newUpstreams, oldUpstreams) && isKnownTransport(newTransport) {
```

Comment:
```suggestion
	if !reflect.DeepEqual(newUpstreams, oldUpstreams) && isKnownTransport(oldTransport) {
```
We want to keep allow the user to do unrecognized things if the _old_ configuration did unrecognized things.